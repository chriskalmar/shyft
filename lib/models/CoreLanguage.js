
import {
  dataTypes,
  Entity,
} from '../../';


const {
  DataTypeString,
} = dataTypes



export const CoreLanguage = new Entity({
  name: 'language',
  domain: 'core',
  description: 'A language',

  // indexing: {
  //   unique: [
  //     [ 'name' ],
  //     [ 'iso_code' ],

  attributes: {

    name: {
      type: DataTypeString,
      description: 'The name of the language',
      maxLength: 30,
      required: true
    },

    nativeName: {
      type: DataTypeString,
      description: 'The native name of the language',
      maxLength: 30,
      required: true
    },

    isoCode: {
      type: DataTypeString,
      description: 'ISO code of the language',
      pattern: '^[a-z]+$',
      minLength: 2,
      maxLength: 2,
      required: true
    },
  }

})

