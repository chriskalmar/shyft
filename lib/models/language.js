
'use script';

module.exports = {

  domain: 'core',

  entities: [

    {
      name: 'language',
      description: 'a language',

      cascadeOnDelete: true,

      attributes: [

        {
          name: 'name',
          type: 'string',
          description: 'Name',
          maxLength: 30,
          required: true
        },

        {
          name: 'native_name',
          type: 'string',
          description: 'Name',
          maxLength: 30,
          required: true
        },

        {
          name: 'iso_code',
          type: 'string',
          description: 'ISO 639-1 code of the language',
          maxLength: 2,
          required: true
        },

      ],

      indexing: {
        unique: [
          [ 'name' ],
          [ 'iso_code' ],
        ]
      }

    }

  ]

}
