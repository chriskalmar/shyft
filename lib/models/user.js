
'use script';

module.exports = {

  domain: 'core',

  entities: [

    {
      name: 'user',
      description: 'a user',

      isUserTable: true,
      cascadeOnDelete: true,

      attributes: [

        {
          name: 'username',
          type: 'string',
          description: 'Username',
          pattern: '^(?![_.])(?!.*[_.]{2})[a-zA-Z0-9._]+$',
          minLength: 3,
          maxLength: 30,
          required: true
        },

        {
          name: 'first_name',
          type: 'string',
          description: 'First name',
          maxLength: 30
        },

        {
          name: 'last_name',
          type: 'string',
          description: 'Last name',
          maxLength: 30
        },

        {
          name: 'language',
          type: 'reference',
          description: 'User language',
          target: 'core::language',
          required: true
        },

        {
          name: 'email',
          type: 'email',
          description: 'Email address',
          required: true
        },

        {
          name: 'password',
          type: 'password',
          description: 'User password',
          required: true
        }
      ],

      indexing: {
        unique: [
          [ 'username' ],
          [ 'email' ],
        ]
      }

    }

  ]

}
