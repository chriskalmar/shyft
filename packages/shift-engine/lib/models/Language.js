
import Entity from '../entity/Entity';
import { DataTypeString } from '../datatype/dataTypes';
import Index, { INDEX_UNIQUE } from '../index/Index';


export const Language = new Entity({
  name: 'Language',
  description: 'A language',

  indexes: [
    new Index({
      type: INDEX_UNIQUE,
      attributes: [ 'name' ],
    }),
    new Index({
      type: INDEX_UNIQUE,
      attributes: [ 'isoCode' ],
    }),
  ],

  attributes: {

    name: {
      type: DataTypeString,
      description: 'The name of the language',
      required: true
    },

    nativeName: {
      type: DataTypeString,
      description: 'The native name of the language',
      required: true
    },

    isoCode: {
      type: DataTypeString,
      description: 'ISO code of the language',
      required: true
    },
  }

})

