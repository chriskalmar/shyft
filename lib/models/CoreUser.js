
import {
  datatypes,
  Entity,
} from '../../';

import CoreLanguage from './CoreLanguage';

const {
  DataTypeString,
} = datatypes

export const CoreUser = new Entity({
  name: 'user',
  domain: 'core',
  description: 'A user',

  // isUserTable: true,
  // sequenceGenerator: 'timebased',

  // indexing: {
  //   unique: [
  //     [ 'username' ],
  //     [ 'email' ],


  attributes: {

    username: {
      type: DataTypeString,
      description: 'Username',
      pattern: '^(?![_.])(?!.*[_.]{2})[a-zA-Z0-9._]+$',
      minLength: 3,
      maxLength: 30,
      required: true
    },

    firstName: {
      type: DataTypeString,
      description: 'First name',
      maxLength: 30
    },

    lastName: {
      type: DataTypeString,
      description: 'Last name',
      maxLength: 30
    },

    language: {
      type: CoreLanguage,
      description: 'User language',
      required: true
    },

    email: {
      type: DataTypeString,
      description: 'Email address',
      required: true
    },

    password: {
      type: DataTypeString,
      description: 'User password',
      required: true
    }
  }

})



