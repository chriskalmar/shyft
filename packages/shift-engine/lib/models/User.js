
import Entity from '../entity/Entity';
import { DataTypeString } from '../datatype/dataTypes';
import Index, { INDEX_UNIQUE } from '../index/Index';

import { Language } from './Language';


export const User = new Entity({
  name: 'User',
  domain: 'core',
  description: 'A user',

  // isUserTable: true,
  // sequenceGenerator: 'timebased',

  indexing: [
    new Index({
      type: INDEX_UNIQUE,
      attributes: [ 'username' ],
    }),
    new Index({
      type: INDEX_UNIQUE,
      attributes: [ 'email' ],
    }),
  ],


  attributes: {

    username: {
      type: DataTypeString,
      description: 'Username',
      pattern: '^(?![_.])(?!.*[_.]{2})[a-zA-Z0-9._]+$',
      minLength: 3,
      maxLength: 30,
      required: true
    },

    firstName: {
      type: DataTypeString,
      description: 'First name',
      maxLength: 30
    },

    lastName: {
      type: DataTypeString,
      description: 'Last name',
      maxLength: 30
    },

    language: {
      type: Language,
      description: 'User language',
      required: true
    },

    email: {
      type: DataTypeString,
      description: 'Email address',
      required: true
    },

    password: {
      type: DataTypeString,
      description: 'User password',
      required: true
    }
  }

})



