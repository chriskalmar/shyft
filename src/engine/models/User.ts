import { Entity } from '../entity/Entity';
import { DataTypeString } from '../datatype/dataTypes';
import { Index, INDEX_UNIQUE } from '../index/Index';

import { Language } from './Language';

export const User = new Entity({
  name: 'User',
  description: 'A user',

  // isUserTable: true,
  // sequenceGenerator: 'timebased',

  indexes: [
    new Index({
      type: INDEX_UNIQUE,
      attributes: [ 'username' ],
    }),
    new Index({
      type: INDEX_UNIQUE,
      attributes: [ 'email' ],
    }),
  ],

  attributes: {
    username: {
      type: DataTypeString,
      description: 'Username',
      required: true,
    },

    firstName: {
      type: DataTypeString,
      description: 'First name',
    },

    lastName: {
      type: DataTypeString,
      description: 'Last name',
    },

    language: {
      type: Language,
      description: 'User language',
      required: true,
    },

    email: {
      type: DataTypeString,
      description: 'Email address',
      required: true,
    },

    password: {
      type: DataTypeString,
      description: 'User password',
      required: true,
    },
  },
});
