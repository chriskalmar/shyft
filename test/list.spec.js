import './setupAndTearDown';
import {
  find,
} from './db';

import {
  asAdmin,
  removeListDynamicData,
} from './testUtils';

import { Profile } from './models/Profile';
import { Board } from './models/Board';
import { Participant } from './models/Participant';


const orderByIdAsc = {
  orderBy: [{
    attribute: 'id',
    direction: 'ASC'
  }]
}

const orderByIdDesc = {
  orderBy: [{
    attribute: 'id',
    direction: 'DESC'
  }]
}

describe('list', () => {

  it('orderBy', async () => {
    const profiles = await find(Profile, { ...orderByIdAsc }, asAdmin())
    profiles.data = removeListDynamicData(Profile, profiles.data)
    expect(profiles).toMatchSnapshot()
  })


  it('orderBy + first', async () => {
    const profiles = await find(Profile, { ...orderByIdAsc, first: 3 }, asAdmin())
    profiles.data = removeListDynamicData(Profile, profiles.data)
    expect(profiles).toMatchSnapshot()
  })


  it('orderBy + first 0', async () => {
    const profiles = await find(Profile, { ...orderByIdAsc, first: 0 }, asAdmin())
    expect(profiles).toMatchSnapshot()
  })


  it('orderBy + last', async () => {
    const profiles = await find(Profile, { ...orderByIdAsc, last: 3 }, asAdmin())
    profiles.data = removeListDynamicData(Profile, profiles.data)
    expect(profiles).toMatchSnapshot()
  })


  it('descending orderBy', async () => {
    const profiles = await find(Profile, { ...orderByIdDesc }, asAdmin())
    profiles.data = removeListDynamicData(Profile, profiles.data)
    expect(profiles).toMatchSnapshot()
  })


  it('descending orderBy + first', async () => {
    const profiles = await find(Profile, { ...orderByIdDesc, first: 3 }, asAdmin())
    profiles.data = removeListDynamicData(Profile, profiles.data)
    expect(profiles).toMatchSnapshot()
  })


  it('descending orderBy + last', async () => {
    const profiles = await find(Profile, { ...orderByIdDesc, last: 3 }, asAdmin())
    profiles.data = removeListDynamicData(Profile, profiles.data)
    expect(profiles).toMatchSnapshot()
  })


  it('orderBy + offset', async () => {
    const profiles = await find(Profile, { ...orderByIdAsc, offset: 5}, asAdmin())
    profiles.data = removeListDynamicData(Profile, profiles.data)
    expect(profiles).toMatchSnapshot()
  })


  it('orderBy + first + offset', async () => {
    const profiles = await find(Profile, { ...orderByIdAsc, first: 3, offset: 5 }, asAdmin())
    profiles.data = removeListDynamicData(Profile, profiles.data)
    expect(profiles).toMatchSnapshot()
  })


  it('orderBy + last + offset', async () => {
    const profiles = await find(Profile, { ...orderByIdAsc, last: 3, offset: 5 }, asAdmin())
    profiles.data = removeListDynamicData(Profile, profiles.data)
    expect(profiles).toMatchSnapshot()
  })


  it('descending orderBy + offset', async () => {
    const profiles = await find(Profile, { ...orderByIdDesc, offset: 5 }, asAdmin())
    profiles.data = removeListDynamicData(Profile, profiles.data)
    expect(profiles).toMatchSnapshot()
  })


  it('descending orderBy + first + offset', async () => {
    const profiles = await find(Profile, { ...orderByIdDesc, first: 3, offset: 5 }, asAdmin())
    profiles.data = removeListDynamicData(Profile, profiles.data)
    expect(profiles).toMatchSnapshot()
  })


  it('descending orderBy + last + offset', async () => {
    const profiles = await find(Profile, { ...orderByIdDesc, last: 3, offset: 5 }, asAdmin())
    profiles.data = removeListDynamicData(Profile, profiles.data)
    expect(profiles).toMatchSnapshot()
  })


  it('filter', async () => {
    const filter = {
      username: 'hazel528'
    }

    const result = await find(Profile, { ...orderByIdAsc, filter }, asAdmin())
    result.data = removeListDynamicData(Profile, result.data)
    expect(result).toMatchSnapshot()
  })


  it('filter with no results', async () => {
    const filter = {
      username: '---not-found---'
    }

    const result = await find(Profile, { ...orderByIdAsc, filter }, asAdmin())
    result.data = removeListDynamicData(Profile, result.data)
    expect(result).toMatchSnapshot()
  })


  it('filter with mutliple attributes', async () => {
    const filter = {
      name: 'Veritatis nihil cum',
      isPrivate: true,
    }

    const result = await find(Board, { ...orderByIdAsc, filter }, asAdmin())
    result.data = removeListDynamicData(Board, result.data)
    expect(result).toMatchSnapshot()
  })


  it('filter with invalid attributes (reject)', async () => {
    const filter = {
      someAttributes: 'Veritatis nihil cum',
    }

    await find(Board, { ...orderByIdAsc, filter }, asAdmin())
      .catch(e => {
        expect(e).toMatchSnapshot()
      })
  })


  describe('filter with logical operators', () => {

    describe('OR', () => {

      const orderBy = [{
        attribute: 'registeredAt',
        direction: 'DESC'
      }]

      const filter = {
        OR: [
          {
            username: 'dana768'
          },
          {
            username: 'weston422'
          },
          {
            username__starts_with: 'jo',
            username__ends_with: '56'
          },
        ]
      }


      it('OR + orderBy', async () => {
        const result = await find(Profile, { orderBy, filter }, asAdmin())
        result.data = removeListDynamicData(Profile, result.data)
        expect(result).toMatchSnapshot()
      })


      it('OR + offset', async () => {
        const result = await find(Profile, { orderBy, filter, offset: 1 }, asAdmin())
        result.data = removeListDynamicData(Profile, result.data)
        expect(result).toMatchSnapshot()
      })


      it('OR + last', async () => {
        const result = await find(Profile, { orderBy, filter, last: 3 }, asAdmin())
        result.data = removeListDynamicData(Profile, result.data)
        expect(result).toMatchSnapshot()
      })

    })


    describe('AND', () => {

      const orderBy = [{
        attribute: 'createdAt',
        direction: 'DESC'
      }]

      const filter = {
        AND: [
          {
            board__gt: 30,
            board__lt: 40
          },
          {
            state__in: [
              20,
              50
            ]
          },
          {
            inviter__in: [
              61,
              78
            ]
          }
        ]
      }


      it('AND + orderBy', async () => {
        const result = await find(Participant, { orderBy, filter }, asAdmin())
        result.data = removeListDynamicData(Participant, result.data)
        expect(result).toMatchSnapshot()
      })


      it('AND + offset', async () => {
        const result = await find(Participant, { orderBy, filter, offset: 4 }, asAdmin())
        result.data = removeListDynamicData(Participant, result.data)
        expect(result).toMatchSnapshot()
      })


      it('AND + last', async () => {
        const result = await find(Participant, { orderBy, filter, last: 4 }, asAdmin())
        result.data = removeListDynamicData(Participant, result.data)
        expect(result).toMatchSnapshot()
      })

    })



    describe('filter with nested logical operators', () => {

      const orderBy = [{
        attribute: 'board',
        direction: 'ASC'
      }]

      const filter = {
        AND: [
          {
            board__gt: 30,
            board__lt: 40
          },
          {
            state__in: [
              20,
              50
            ]
          },
          {
            inviter__in: [
              61,
              78
            ]
          },
          {
            OR: [
              {
                invitee: 107
              },
              {
                invitee: 21
              },
              {
                invitee: 38
              },
              {
                invitee__gt: 60,
                invitee__lt: 62
              }
            ]
          }
        ]
      }


      it('AND + OR', async () => {
        const result = await find(Participant, { orderBy, filter }, asAdmin())
        result.data = removeListDynamicData(Participant, result.data)
        expect(result).toMatchSnapshot()
      })

    })

  })


  it('parentConnection', async () => {
    const parentConnection = {
      id: 60,
      attribute: 'inviter'
    }

    const result = await find(Participant, { ...orderByIdAsc }, asAdmin(), parentConnection)
    result.data = removeListDynamicData(Participant, result.data)
    expect(result).toMatchSnapshot()
  })


  it('parentConnection + first', async () => {
    const parentConnection = {
      id: 60,
      attribute: 'inviter'
    }

    const result = await find(Participant, { ...orderByIdAsc, first: 1 }, asAdmin(), parentConnection)
    result.data = removeListDynamicData(Participant, result.data)
    expect(result).toMatchSnapshot()
  })


  it('parentConnection with invalid attribute (reject)', async () => {
    const parentConnection = {
      id: 60,
      attribute: 'invalidAttributeName'
    }

    await await find(Participant, { ...orderByIdAsc }, asAdmin(), parentConnection)
      .catch(e => {
        expect(e).toMatchSnapshot()
      })
  })



})
