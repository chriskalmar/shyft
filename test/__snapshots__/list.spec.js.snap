// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`list descending orderBy + first 1`] = `
Object {
  "data": Array [
    Object {
      "confirmedAt": null,
      "id": "110",
      "password": "e97188c43aaf749ffa43df9286c35eb150174bb7bfe69dc6df7689fb782b976a",
      "username": "krystel170",
    },
    Object {
      "confirmedAt": null,
      "id": "109",
      "password": "bf94deb712f9040ba1effa106091e0a1ca330f4b1f311e44b021e955ad203100",
      "username": "muriel707",
    },
    Object {
      "confirmedAt": null,
      "id": "108",
      "password": "b00046da5a1751e46b557fab4edb5d91e13c44c8a0205fc2d89275d422df49ad",
      "username": "tianna226",
    },
  ],
  "pageInfo": Object {
    "hasNextPage": true,
    "hasPreviousPage": false,
  },
}
`;

exports[`list descending orderBy + last 1`] = `
Object {
  "data": Array [
    Object {
      "confirmedAt": null,
      "id": "3",
      "password": "d36dc0a7b548fe7a1e0da69049ea1b14db66b6ea408e297c8f700e666843b020",
      "username": "fleta342",
    },
    Object {
      "confirmedAt": null,
      "id": "2",
      "password": "0ba4c26f03ae3c7c7a85bf4204539884e52a0e1e511aeaf11d3b2fd627b5957d",
      "username": "anya543",
    },
    Object {
      "confirmedAt": null,
      "id": "1",
      "password": "8fed8061e1e33d96169b9026605172a8fe21eb1ce51756187195be0330536fb7",
      "username": "annabel711",
    },
  ],
  "pageInfo": Object {
    "hasNextPage": false,
    "hasPreviousPage": true,
  },
}
`;

exports[`list descending orderBy 1`] = `
Object {
  "data": Array [
    Object {
      "confirmedAt": null,
      "id": "110",
      "password": "e97188c43aaf749ffa43df9286c35eb150174bb7bfe69dc6df7689fb782b976a",
      "username": "krystel170",
    },
    Object {
      "confirmedAt": null,
      "id": "109",
      "password": "bf94deb712f9040ba1effa106091e0a1ca330f4b1f311e44b021e955ad203100",
      "username": "muriel707",
    },
    Object {
      "confirmedAt": null,
      "id": "108",
      "password": "b00046da5a1751e46b557fab4edb5d91e13c44c8a0205fc2d89275d422df49ad",
      "username": "tianna226",
    },
    Object {
      "confirmedAt": null,
      "id": "107",
      "password": "f735b74ca8ad9b0f7ae8190fa71acfa15f6a5a4f5b7a8ac2ef790c59eb185e3c",
      "username": "leanna128",
    },
    Object {
      "confirmedAt": null,
      "id": "106",
      "password": "507b91e6b0ae6ea80e7c0d1642304ce8cd713527d36db297fef1504598ecd79c",
      "username": "stan321",
    },
    Object {
      "confirmedAt": null,
      "id": "105",
      "password": "bac53a2cdcbf0a3d8fc1b1e573e81699382ed1080821d20ec1b47c7dddd02fa5",
      "username": "brandi646",
    },
    Object {
      "confirmedAt": null,
      "id": "104",
      "password": "8fed8061e1e33d96169b9026605172a8fe21eb1ce51756187195be0330536fb7",
      "username": "joe154",
    },
    Object {
      "confirmedAt": null,
      "id": "103",
      "password": "f056b8cae37dc3ed60ee52da9a2951f8a6c531fe3222a3ad6835dbdd905c7d2a",
      "username": "ella377",
    },
    Object {
      "confirmedAt": null,
      "id": "102",
      "password": "e97188c43aaf749ffa43df9286c35eb150174bb7bfe69dc6df7689fb782b976a",
      "username": "garfield347",
    },
    Object {
      "confirmedAt": null,
      "id": "101",
      "password": "e44591879052d99a7509f498e6e65985123b34c9b69ff6eafacb9b05f9deb1eb",
      "username": "isidro418",
    },
  ],
  "pageInfo": Object {
    "hasNextPage": true,
    "hasPreviousPage": false,
  },
}
`;

exports[`list orderBy + first 1`] = `
Object {
  "data": Array [
    Object {
      "confirmedAt": null,
      "id": "1",
      "password": "8fed8061e1e33d96169b9026605172a8fe21eb1ce51756187195be0330536fb7",
      "username": "annabel711",
    },
    Object {
      "confirmedAt": null,
      "id": "2",
      "password": "0ba4c26f03ae3c7c7a85bf4204539884e52a0e1e511aeaf11d3b2fd627b5957d",
      "username": "anya543",
    },
    Object {
      "confirmedAt": null,
      "id": "3",
      "password": "d36dc0a7b548fe7a1e0da69049ea1b14db66b6ea408e297c8f700e666843b020",
      "username": "fleta342",
    },
  ],
  "pageInfo": Object {
    "hasNextPage": true,
    "hasPreviousPage": false,
  },
}
`;

exports[`list orderBy + last 1`] = `
Object {
  "data": Array [
    Object {
      "confirmedAt": null,
      "id": "108",
      "password": "b00046da5a1751e46b557fab4edb5d91e13c44c8a0205fc2d89275d422df49ad",
      "username": "tianna226",
    },
    Object {
      "confirmedAt": null,
      "id": "109",
      "password": "bf94deb712f9040ba1effa106091e0a1ca330f4b1f311e44b021e955ad203100",
      "username": "muriel707",
    },
    Object {
      "confirmedAt": null,
      "id": "110",
      "password": "e97188c43aaf749ffa43df9286c35eb150174bb7bfe69dc6df7689fb782b976a",
      "username": "krystel170",
    },
  ],
  "pageInfo": Object {
    "hasNextPage": false,
    "hasPreviousPage": true,
  },
}
`;

exports[`list orderBy 1`] = `
Object {
  "data": Array [
    Object {
      "confirmedAt": null,
      "id": "1",
      "password": "8fed8061e1e33d96169b9026605172a8fe21eb1ce51756187195be0330536fb7",
      "username": "annabel711",
    },
    Object {
      "confirmedAt": null,
      "id": "2",
      "password": "0ba4c26f03ae3c7c7a85bf4204539884e52a0e1e511aeaf11d3b2fd627b5957d",
      "username": "anya543",
    },
    Object {
      "confirmedAt": null,
      "id": "3",
      "password": "d36dc0a7b548fe7a1e0da69049ea1b14db66b6ea408e297c8f700e666843b020",
      "username": "fleta342",
    },
    Object {
      "confirmedAt": null,
      "id": "4",
      "password": "94b5f0fa59e958580234a46304f711dbf0276bcee26ce8ef14e9c750fcaa5a2b",
      "username": "victor821",
    },
    Object {
      "confirmedAt": null,
      "id": "5",
      "password": "7679f40138809eb3e012b29a46160f797b3afd72227099bde320fbc7e725e59d",
      "username": "griffin807",
    },
    Object {
      "confirmedAt": null,
      "id": "6",
      "password": "bbd72dedbd01e6593f9bdb72cb6372f299c686e922fc992aafbe820985dce209",
      "username": "jeremy386",
    },
    Object {
      "confirmedAt": null,
      "id": "7",
      "password": "a71d5794449a7019156a0a06fccf3e5ef522fc8767edee6b08292050c52c25d3",
      "username": "autumn520",
    },
    Object {
      "confirmedAt": null,
      "id": "8",
      "password": "5f53b0a2be22e28dc4b056f663910c046a33f9b2d779eaf997ea771fe1881a13",
      "username": "edward969",
    },
    Object {
      "confirmedAt": null,
      "id": "9",
      "password": "913a5aa81861da1fe643478d72c343152b38d0fe841a6b02bbe74bab7004a2de",
      "username": "omari409",
    },
    Object {
      "confirmedAt": null,
      "id": "10",
      "password": "16149626b0365e04a9a52d02e17ca8aeb63680b9f5ee2ba334b0b1a38a833cfe",
      "username": "leon807",
    },
  ],
  "pageInfo": Object {
    "hasNextPage": true,
    "hasPreviousPage": false,
  },
}
`;
